### ISSUE TITLE
Wheels for Python 3.13

### ISSUE THREAD

**Issue by mcgrawia** [apache/datasketches-python#54]
Hi Datasketch team, 

My team and I were hoping to upgrade to Python 3.13, but it doesn't look like there are wheels available for 3.13 yet. Is it possible to kick off a build job to publish wheels for 3.13? 

I'd be happy to assist if there's anything I can do.

Thanks!

**Comment by AlexanderSaydakov** [comment]
for what platform?

**Comment by AlexanderSaydakov** [comment]
I run the "Build Python Wheels" action for the latest code from the main branch, and I see that it builds the following:
- 3.12.7 for Windows
- 3.13.0 for MacOS
- 3.12.7 for Ubuntu

**Comment by mcgrawia** [comment]
From what I'm seeing in the job logs, it looks like it's running the build in the following container image:
```
Unable to find image 'quay.io/pypa/manylinux2014_x86_64:2024-01-23-12ffabc' locally
```
Is it possible to change the tag that build image is using? It looks like that image is from January and that was before 3.13 was available. It looks like the latest tag is `2024-11-30-fa0d298` from a few days ago? 

https://quay.io/repository/pypa/manylinux2014_x86_64?tab=tags&tag=latest

**Comment by mcgrawia** [comment]
I don't have access to see it, but I'm guessing we may need to bump the `cibuildwheel` version used? 

https://github.com/apache/datasketches-python/actions/runs/12168116607/workflow#L96

```
      - name: Install Python dependencies
        run: python -m pip install cibuildwheel~=${{ env.MIN_CIBUILDWHEEL_VERSION }}
```

**Comment by jmalkin** [comment]
Ugh, annoying that it doesn't take a later version if available.

I created a discussion on our slack channel about whether we can do this under Apache policy without a formal release.

**Comment by mcgrawia** [comment]
thanks @jmalkin! I just created an MR in https://github.com/apache/datasketches-python/pull/55 to bump the version if that helps. Let me know

**Comment by jmalkin** [comment]
Merging auto-closed, but the underlying issue is not yet resolved.

So...that earlier build is still going, and it's all cuz of muslinux on arm64.  Transpiling for muslinux is very, very, very slow. We might need to cut off the oldest version (3.7?) at least for that platform to add another one and ensure the job can finish within GHA's per-task time allowance!

**Comment by jmalkin** [comment]
Nope, already cut off 3.7.  I guess we can kill off 3.8 as that's officially EOLed as of last month.

**Comment by mcgrawia** [comment]
thanks @jmalkin. Do we need to remove 3.8 before we can run another build? If so, happy to create an MR for that as well. Let me know

**Comment by jmalkin** [comment]
Running the job now, but unfortunately the version config is in pyproject.toml which is part of the released code. So I think that's technically a patch release, which requires our usual release process.

**Comment by mcgrawia** [comment]
Ok thanks @jmalkin. Looks like with the new version of the `cibuildwheel`, the new cross compile builds only take about 8 minutes each instead of 50 minutes. The whole job finished in 40 minutes instead of 4 hours! Makes sense that the changes to `pyproject.toml` would require a release. Curious, would it be possible to kick off a build/publish job that just has the `cibuildwheel` upgrade without the `pyproject.toml` changes? Since the compilation for musl runs much faster now, I think it should still run ok within the time limit to include 3.8. Not sure if that approach would allow you to publish a 3.13 wheel without having to go through the release process again just for this issue?

**Comment by jmalkin** [comment]
@AlexanderSaydakov and I debated it and since we do include the wheels as part of what gets voted on and approved with the release, it feels like a process violation to have new wheels built from a different commit hash.

Especially since the wheels aren't auto-pushed directly from the github action and transit through one of the developer's computer it seems good to ensure that the gpg signature and shasum can be matched against the release artifact.

There's debate within the Apache community on how to streamline the release process, and I'm very much in favor of that. Being able to trigger deployments from workflows would be wonderful! But those discussions seem to be moving along very slowly. In the meantime, the process is designed both to protect the Apache organization but, equally importantly, to make it hard to rush out changes without proper review. The hope is to avoid an XZ Utils type situation where a rogue developer (or group) took over a critical utility. So I very much appreciate the intent and try to work within the confines of the release process even though there are times like this where it can be quite annoying.

**Comment by mcgrawia** [comment]
Thanks for the explanation @jmalkin. I am not familiar with the apache process, so that was helpful context. I'll keep an eye out for the next release then for the new wheels. Thanks for the help!

**Comment by jmalkin** [comment]
@mcgrawia  This should now be released and available from pypi as 5.1.1. Thanks for your patience with the process.

**Comment by mcgrawia** [comment]
thanks @jmalkin! Confirming the new wheels are working in our builds. Appreciate the help!
